### ISSUE TITLE
How to format dates with a certain locale ?

### ISSUE THREAD

**Issue by iulian-olteanu** [relair/material-dynamic-table#20]
Hello,

Not an issue, but more like a question or suggestion.

I would like to use this throughout an app which handles loads of data.
Is there any way to format dates in another locales than en-US ? 
Also, can this be a consistent setting (for example -> set it once and be valid throughout the entire app) ?

For example, in the material.module.ts, one has the option to use MAT_DATE_LOCALE and then make use of useValue in the providers array, thus offering a consistent user experience with datepicker for example.

import { MAT_DATE_LOCALE } from '@angular/material/core';

@NgModule({
imports: [],
exports: [],
providers: [{ provide: MAT_DATE_LOCALE, useValue: '<locale>' }],
  declarations: [],
})

I do not know if it's worth considering since once can do -> dateFormat: 'dd.MM.yyyy' for example.

**Comment by iulian-olteanu** [comment]
Nevermind

**Comment by RoyChase** [comment]
I have a similar question.. What was the resolution if any?

**Comment by iulian-olteanu** [comment]
Formatting dates is related to Angular. In my case the locale is RO. I have set it up globally for the app I am working on.

Using this I can define a date column like this:
{
      name: '<name>',
      displayName: '<display name>,
      type: 'date',
      hidden: false,
      options: {
        dateFormat: 'dd.MM.yyyy',
      },
    },
Or
the default (short) which gives me dates such as: dd.MM.yyyy HH:mm (example: 29.07.2020, 13:19).

This is how I did it:

- I used registerLocaleData in my app.module.ts and used LOCALE_ID

1)  import { registerLocaleData } from '@angular/common';
     import localeRO from '@angular/common/locales/ro';
    import { LOCALE_ID } from '@angular/core';
2) registerLocaleData(localeRO);
3) In app.module.ts, in the providers array I added
{ provide: LOCALE_ID, useValue: 'ro' } (  )

- In material.module.ts, I modified the providers and used MomentJS


import {
  MatNativeDateModule,
  NativeDateModule,
  MAT_DATE_FORMATS,
  NativeDateAdapter,
  DateAdapter, MatCommonModule
} from '@angular/material/core';

import { MAT_DATE_LOCALE } from '@angular/material/core';
import {
  MatMomentDateModule,
  MomentDateModule,
  MomentDateAdapter,
  MAT_MOMENT_DATE_ADAPTER_OPTIONS,
} from '@angular/material-moment-adapter';
import { MatDividerModule } from '@angular/material/divider';

export const MY_FORMATS = {
  parse: {
    dateInput: 'DD.MM.YYYY',
  },
  display: {
    dateInput: 'DD.MM.YYYY',
    monthYearLabel: 'MMMM YYYY',
    dateA11yLabel: 'LLL',
    monthYearA11yLabel: 'MMMM YYYY',
  },
};


  providers: [
    { provide: MAT_DATE_LOCALE, useValue: 'ro' },
    { provide: MAT_DATE_FORMATS, useValue: MY_FORMATS },
    {
      provide: DateAdapter,
      useClass: MomentDateAdapter,
      deps: [MAT_DATE_LOCALE, MAT_MOMENT_DATE_ADAPTER_OPTIONS],
    },
